_source/_posts/2022-01-27-build-website-with-eleventy.md

```diff
@@ -0,0 +1,334 @@
+
+Be sure to replace the placeholder variables with your actual Okta information.
+
+This file will be the base layout for all of your pages.  This file loads `bootstrap` into all of your pages.  It also initializes the `OktaSignIn` widget on each page.  This way, you can access it in the javascript of your file no matter where you are.
+
+Next, add a file for `_layout_unauthenticated.11ty.js` and add the following code to it.
+
+```javascript
+exports.data = {
+  title: "11ty And Okta",
+  layout:  "_layout_base.11ty.js"
+};
+
+exports.render = function (data) {
+  return `
+     <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">
+      <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
+       Home
+      </a>
+
+      <div class="col-md-3 text-end">
+        <button type="button" class="btn btn-outline-primary me-2"><a href="Login">Login</a></button>
+      </div>
+    </header>
+
+    ${data.content}
+        
+        `;
+};
+```
+
+This file extends the base layout you created by creating a header with a `Home` button and a `Login` button that will direct the user to the `Login` page with your Okta sign-in widget.
+
+Finally, add a new file called `_layout_authenticated.11ty.js`. Add the following code to it.
+
+```javascript
+exports.data = {
+title: "11ty And Okta",
+layout: "_layout_base.11ty.js",
+};
+
+exports.render = function (data) {
+return `
+
+<header
+    class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">
+    <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
+        Home
+    </a>
+
+    <div class="col-md-3 text-end">
+        <button type="button" class="btn btn-outline-primary me-2" onclick='logout()'>Logout</button>
+    </div>
+</header>
+
+${data.content}
+
+<script type="text/javascript">
+    function logout() {
+        oktaSignIn.authClient.signOut();
+        location.reload();
+    }
+
+    if(!oktaSignIn?.authClient?.token) {
+        window.location='/Login';
+    }
+
+    oktaSignIn.authClient.token.getUserInfo().then(function(user) {
+        if(loadPage) {
+            loadPage();
+        }
+    }).catch(function(e) {
+        console.log(e);
+        window.location='/Login'
+    });
+</script>
+`;
+};
+```
+
+Again, this file will extend the base layout. It also adds some additional functionality to each page using this layout. When the page loads, it will attempt to contact the Okta authorization server to obtain information about the user with the ID token stored in the browser. If that request is successful, it will call the `loadPage` function implemented in those pages. Otherwise, it redirects the user to the `Login` page. This page also contains a header with a `Logout` button and a method for `logout` that uses the sign-in widget to log the user out.
```

Comment:
```suggestion
Again, this file will extend the base layout, while adding some additional functionality to each page that uses this layout. When the page loads, it will attempt to contact the Okta authorization server for information about the user with the ID token stored in the browser. If the contact request is successful, it will call the `loadPage` function implemented in those pages. Otherwise, the code redirects the user to the `Login` page. This page also contains a header with a `Logout` button and a method for `logout` that uses the sign-in widget to log the user out.
```
attempts to clarify what "it" refers to, and simplify the sequence of steps. Please verify for accuracy